test/reductions/grid_search/test_grid_search_arguments.py

```diff
@@ -0,0 +1,290 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import logging
+import numpy as np
+import pandas as pd
+import pytest
+from sklearn.linear_model import LogisticRegression, LinearRegression
+
+from fairlearn.metrics import DemographicParity, BoundedGroupLoss
+from fairlearn.reductions import GridSearch
+from fairlearn.reductions.grid_search.simple_quality_metrics import SimpleClassificationQualityMetric, SimpleRegressionQualityMetric  # noqa: E501
+
+# ==============================================================
+
+# The following are functions which convert ndarrays into other datatypes
+# They are used to generate different argument types for calls to
+# GridSearch
+
+
+def identity(X):
+    return X
+
+
+def pandasdf(X):
+    return pd.DataFrame(X)
+
+
+def pandasseries(X):
+    # Will not work if ndarray has more than one dimension
+    return pd.Series(X)
+
+
+def ndarray2d(X):
+    # Adds a second dimension of length 1 onto a 1D
+    # array. This is for checking that shapes n and
+    # n*1 behave the same
+    if len(X.shape) != 1:
+        raise RuntimeError("ndarray2d requires 1d ndarray")
+
+    X.shape = (X.shape[0], 1)
+    return X
+
+
+# List the different datatypes which need to succeed for
+# all GridSearch calls
+Xtransform = [identity, pandasdf]
+Ytransform = [identity, pandasdf, pandasseries, ndarray2d]
+Atransform = [identity, pandasdf, pandasseries, ndarray2d]
+
+
+# Base class for tests
+# Tests which must be passed by all calls to the GridSearch
+# go here
+class ArgumentTests:
+    def _quick_data(self, number_samples=48):
+        feature_1 = np.random.randint(2, size=number_samples)
```

Comment:
if your data is randomized how can we reproduce test failures?

Reply:
I'm using random data to make it plain that the actual values don't matter. We can get occasional failures if all the labels have the same value - that's why I bumped the number of samples up to 48.

Reply:
Sure, none of the values matter in tests, but say it fails for some weird combination, then we can't even figure out what it was. Rerunning it might fix it then, but we've lost the combination.

Reply:
You could fix the seed, or parametrize the seed as well with a bunch of different values.

Reply:
Have hard coded some carefully selected random numbers